### PR TITLE
[DOC] Major documentation update

### PR DESCRIPTION
**Pull request by pvchupin**
* Simplified instructions for the first local run
* Clarified variables, parameters meaning
* Fixed inconsistencies with implementation
* Outlined special test categories External and ExtraTests
* Other minor fixes

TODO:
* Update LIT parameters, some are not used now, some are not documented

Signed-off-by: Pavel V Chupin <pavel.v.chupin@intel.com>

### CODE REVIEW

File: SYCL/ExtraTests/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Overview
+
+"ExtraTests" directory contains infrastructure for picking up LIT style tests
+from another external directory passed in SYCL_EXTRA_TESTS_SRC.
+
+# Cmake parameters
```
Comment (vladimirlaz) [inline SYCL/ExtraTests/README.md]:
```suggestion
# CMake parameters
```

**Review by vladimirlaz — DISMISSED** [review]

**Review by vladimirlaz — DISMISSED** [review]

File: SYCL/External/README.md [hunk 2]
```diff
@@ -1,82 +1,75 @@
...
-Notes:
- - it is assumed that LIT framework, FileCheck and other LIT dependencies are
-available in the same directory with llvm-lit.
+"External" tests directory has necessary infrastructure for running tests where
+sources of tests are expected to be located outside of llvm-test-suite
+repository. Each subdirectory represents external repository for corresponding
+application/test.
 
 # Directory structure
 Every sub-directory (e.g. RSBench) contains the following content:
```
Comment (bader) [inline SYCL/External/README.md]: What is `RSBench`? Can we add a link to this benchmark?

File: SYCL/External/README.md [hunk 3]
```diff
@@ -1,82 +1,75 @@
...
  * **CMakeLists.txt** - CMake configuration file which is used to obtain
- application binary and data files. There are several variables which are
- used by it (see [Main parameters](#main-parameters) section for details)
+ application binary and data files.
  * **\*.test** - test configuration files, containing command lines, test
-   status lines and verification patterns following [LLVM test infrastructure](https://llvm.org/docs/TestingGuide.html).
+   status lines and verification patterns following
+   [LLVM tests infrastructure](https://llvm.org/docs/TestingGuide.html).
```
Comment (bader) [inline SYCL/External/README.md]: Why is this required? My understanding is that only **lit.local.cfg** is required and this file configures now external benchmark is tested. As you can see below the file name extension is defined by this file and can be something different (e.g. `*.cpp`).

File: SYCL/ExtraTests/README.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# Overview
+
+"ExtraTests" directory contains infrastructure for picking up LIT style tests
+from another external directory passed in SYCL_EXTRA_TESTS_SRC.
```
Comment (bader) [inline SYCL/ExtraTests/README.md:8]: Could you clarify what is the difference with [External](https://github.com/intel/llvm-test-suite/tree/intel/SYCL/External) tests, please?
It looks to me that both are intended to run LIT-based tests located out-of-tree. Right?

File: SYCL/README.md [hunk 5]
```diff
@@ -1,175 +1,167 @@
...
-   from external directory passed in SYCL_EXTRA_TESTS_SRC. 
-
-# Execution
+This directory contains SYCL-related tests distributed in subdirectories based
+on testing scope.
+
+# Prerequisites
+
+ - DPC++ compiler. Can be built following these
+   [instructions](https://github.com/intel/llvm/blob/sycl/sycl/doc/GetStartedGuide.md#build-dpc-toolchain)
```
Comment (bader) [inline SYCL/README.md:18]: Shouldn't we recommend to take pre-built binaries instead of asking to build it from sources? E.g. Contributing guide describes CI checks taking nightly builds.

File: SYCL/External/README.md (hunk 2 above)
Comment (pvchupin) [reply SYCL/External/README.md]: Sure. Added.

File: SYCL/External/README.md (hunk 3 above)
Comment (pvchupin) [reply SYCL/External/README.md]: I think you are right in general and probably depends on case. It's just how RSBench example has been integrated. This approach is also documented at https://llvm.org/docs/TestingGuide.html#platform-specific-tests.
I've rephrased this section to focus more on RSBench as an example, rather than set of real requirements. Other cases may or may not needed to reuse these.

Comment (pvchupin) [reply SYCL/External/README.md]: Also I'm not sure if RSBench test is functional at all. It looks it's marked as XFAIL for all platforms. I don't even see it downloading sources when I tried to run on host. @vladimirlaz, can you comment on that? Is that a placeholder of some sort?

File: SYCL/ExtraTests/README.md (hunk 4 above)
Comment (pvchupin) [reply SYCL/ExtraTests/README.md:8]: It seems ExtraTests run tests which are completely under LIT framework (e.g. in directory elsewhere) and External tests are to run other frameworks. That's my guess.
@vladimirlaz, can you confirm? What was the purpose and do we use any of these right now?

File: SYCL/README.md (hunk 5 above)
Comment (pvchupin) [reply SYCL/README.md:18]: Updated text.

File: SYCL/ExtraTests/README.md (hunk 4 above)
Comment (vladimirlaz) [reply SYCL/ExtraTests/README.md:8]: @pvchupin you are right about ExtraTests and External test.
 - ExtraTests can be used to run intel/llvm in-tree LIT tests from llvm-test-suite framework. This can be removed I think.
 - External tests allow to download/build/run external applications (e.g. RSBench from https://github.com/ANL-CESAR/RSBench.git). The tests were marked as XFail because benchmark internal data verification failed (it is successfully build and code is executed without crashes). There is no current need for that as well.

**Review by bader — DISMISSED** [review]

File: SYCL/ExtraTests/README.md (hunk 4 above)
Comment (pvchupin) [reply SYCL/ExtraTests/README.md:8]: Marked both as experimental. We should probably remove at some later point if no use.

**Review by vladimirlaz — APPROVED** [review]
